modules/whatshap/1.0/whatshap.smk

```diff
@@ -0,0 +1,195 @@
+                "\n" f"ERROR: This module requires oncopipe version >= {min_oncopipe_version}. Please update oncopipe in your environment" + '\x1b[0m'
+                )
+    sys.exit("Instructions for updating to the current version of oncopipe are available at https://lcr-modules.readthedocs.io/en/latest/ (use option 2)")
+
+# End of dependency checking section 
+
+# Setup module and store module-specific configuration in `CFG`
+# `CFG` is a shortcut to `config["lcr-modules"]["whatshap"]`
+CFG = op.setup_module(
+    name = "whatshap",
+    version = "1.0",
+    subdirectories = ["inputs", "clair3", "filter_clair3", "whatshap_phase_vcf", "whatshap_phase_bam" ,"outputs"]
+)
+
+# Define rules to be run locally when using a compute cluster
+localrules:
+    _promethion_input,
+    _filter_clair3,
+    _whatshap_all
+
+
+##### RULES #####
+
+
+# Symlinks the input files into the module results directory (under '00-inputs/')
+rule _promethion_input:
+    input:
+        bam = CFG["inputs"]["sample_bam"],
+        bai = CFG["inputs"]["sample_bai"]
+    output:
+        bam = CFG["dirs"]["inputs"] + "bam/{seq_type}--{genome_build}/{sample_id}.bam",
+        bai = CFG["dirs"]["inputs"] + "bam/{seq_type}--{genome_build}/{sample_id}.bam.bai"
+    run:
+        op.absolute_symlink(input.bam, output.bam)
+        op.absolute_symlink(input.bai, output.bai)
+        
+
+rule _clair3:
+    input:
+        bam = CFG["dirs"]["inputs"] + "bam/{seq_type}--{genome_build}/{sample_id}.bam",
+        fasta = reference_files("genomes/{genome_build}/genome_fasta/genome.fa")
+    params: 
+        model = CFG["clair3"]["model"],
+        dir = CFG["dirs"]["clair3"] + "{seq_type}--{genome_build}/{sample_id}",
+        platform = CFG["clair3"]["platform"]
+    conda :
+        CFG["conda_envs"]["clair3"]  
+    threads:
+        CFG["threads"]["clair3"]    
+    resources: 
+        mem_mb = CFG["mem_mb"]["clair3"]     
+    log:
+        stderr = CFG["logs"]["clair3"] + "{seq_type}--{genome_build}/{sample_id}/clair3.stderr.log"    
+    output:
+        vcf = CFG["dirs"]["clair3"] + "{seq_type}--{genome_build}/{sample_id}/phased_merge_output.vcf.gz"          
+    shell:
+         op.as_one_line("""run_clair3.sh  
+            -b {input.bam} -f {input.fasta} -t {threads} -p {params.platform}
+            -m {params.model} -o {params.dir}
+            --remove_intermediate_dir --enable_phasing
+            2> {log.stderr} """)
+
+rule _filter_clair3:
+    input:
+        vcf = str(rules._clair3.output.vcf)
+    output: 
+        filtered = CFG["dirs"]["filter_clair3"] + "{seq_type}--{genome_build}/{sample_id}.filtered_phased_merged.vcf.gz"
+    conda :
+        CFG["conda_envs"]["clair3"]
+    log:
+        stderr = CFG["logs"]["filter_clair3"] + "{seq_type}--{genome_build}/{sample_id}/filter_clair3.stderr.log"      
+    shell :
+        "zcat {input.vcf} | awk '$6 > 20 || $1 ~ /^#/' | bgzip > {output.filtered} && tabix -p vcf {output.filtered}  2> {log.stderr}"       
```

Comment:
Ditto here - should this filtering threshold be user-configurable?

Reply:
This has been moved to phase_vcf module and fixed there. Whatshap module now takes phased vcf as input.